Source/EmbeddedBoundary/Enabled.cpp

```diff
@@ -0,0 +1,45 @@
+/* Copyright 2024 Axel Huebl
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#include "Enabled.H"
+
+#ifdef AMREX_USE_EB
+#include <AMReX_ParmParse.H>
+#endif
+#if defined(AMREX_USE_EB) && defined(WARPX_DIM_RZ)
+#include <stdexcept>
+#endif
+
+
+namespace EB
+{
+    bool enabled ()
+    {
+#ifndef AMREX_USE_EB
+        return false;
+#else
+        amrex::ParmParse const pp_warpx("warpx");
+        amrex::ParmParse const pp_eb2("eb2");
+
+        // test various runtime options to enable EBs
+        std::string eb_implicit_function;
+        bool eb_enabled = pp_warpx.query("eb_implicit_function", eb_implicit_function);
+
+        // https://amrex-codes.github.io/amrex/docs_html/EB.html
+        std::string eb_stl;
+        eb_enabled |= pp_eb2.query("geom_type", eb_stl);
+
+#if defined(WARPX_DIM_RZ)
+        if (eb_enabled) {
+            throw std::runtime_error("RZ Geometry does not yet support EBs, but EBs are enabled in runtime inputs.");
+        }
+#endif
+
```

Comment:
@RemiLehe told me on Slack that we have RZ support for EBs with test in ES and EM. So this can go :tada: 
```suggestion
```